plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/launcher/PipelineLaunchConfiguration.java

```diff
@@ -122,20 +123,44 @@ public void setUseDefaultLaunchOptions(boolean useDefaultLaunchOptions) {
     this.useDefaultLaunchOptions = useDefaultLaunchOptions;
   }
 
-  public String getEclipseProjectName() {
-    return eclipseProjectName.orNull();
+  public MajorVersion getMajorVersion() {
+    return majorVersion;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    } else if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+    PipelineLaunchConfiguration other = (PipelineLaunchConfiguration) obj;
+    return Objects.equals(majorVersion, other.majorVersion)
+        && Objects.equals(argumentValues, other.argumentValues)
+        && Objects.equals(runner, other.runner)
+        && Objects.equals(userOptionsName, other.userOptionsName);
```

Comment:
Should we also compare the `useDefaultLaunchOptions` field?

Reply:
It should.  I *suspect* I didn't include this as held as have another PR (that needs dusting off) that removes this support (from [#2430](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2430#issuecomment-336514168)).